website/docs/core-concepts/components/component-inheritance.md

```diff
@@ -4,11 +4,254 @@ sidebar_position: 7
+- **Principle of abstraction**: In the `ue2-dev` stack, only the relevant information about the derived components in the stack is shown. All the base
+  component settings are "hidden" (in the imported `catalog`), which reduces the configuration size and complexity
+- **Dynamic polymorphism**: The derived `vpc-1` and `vpc-2` components override and use the base component properties to be able to provision the same
+  Terraform configuration many times but with different settings
+
+<br/>
+
+Having the components in the stack configured as shown above, we can now provision the `vpc-1` and `vpc-2` components into the `ue2-dev` stack by
+executing the following `atmos` commands:
+
+```shell
+atmos terraform apply vpc-1 -s ue2-dev
+atmos terraform apply vpc-2 -s ue2-dev
+```
+
+<br/>
+
+As we can see, using the principles of **Component-Oriented Programming (COP)**, we are able to define two (or more) components with
+different settings, and provision them into the same (or different) environment (account/region) using the same Terraform code (which is
+environment-agnostic). And the configurations are extremely DRY and reusable.
+
+## Multiple Inheritance
+
+Multiple Inheritance is used when an Atmos component inherits from more than one base components.
+
+Multiple Inheritance allows a component to inherit from many base components or mixins, each base component having its own inheritance chain,
+effectively making it an inheritance matrix. It uses a method similar to Method Resolution Order (MRO) using
+the [C3 linearization](https://en.wikipedia.org/wiki/C3_linearization) algorithm, which is how Python supports multiple inheritance.
+
+<br/>
+
+:::info
+
+In **Object-Oriented Programming (OOP)**, a mixin is a class that contains methods for use by other classes without having to be the parent class of
+those other classes.
+
+In **Component-Oriented Programming (COP)** implemented by Atmos, a mixin is an abstract base component that is never meant to be provisioned and does
+not have any physical implementation - it just contains default settings/variables/properties for use by other Atmos components.
+
+:::
+
+<br/>
+
+Multiple Inheritance, similarly to Single Inheritance, is defined by the `metadata.inherits` section in the component
+configuration. `metadata.inherits` is a list of component or mixins names from which the current component inherits.
+In the case of multiple base components, it is processed in the order by which it was declared.
+
+For example, in the following configuration:
+
+```yaml
+metadata:
+  inherits:
+    - componentA
+    - componentB
+```
+
+Atmos will recursively deep-merge all the base components of `componentA` (each component overriding its base),
+then all the base components of `componentB` (each component overriding its base), then the two results are deep-merged together with `componentB`
+inheritance chain overriding the values from `componentA` inheritance chain.
+
```

Comment:
:::important
All the base components referenced by `inherits` must be defined in the Stack configuration, either by using `import` or by explicitly defining them in the Stack configuration
:::